### ISSUE TITLE
TypeError: 'NoneType' object is not iterable in discovery

### ISSUE THREAD

**Issue by nmxprime** [SoCo/SoCo#668]
I have recently upgraded to Release version 0.17 using pip upgrade
`pip install soco --upgrade`
After upgrade, the example `play_local_files.py` throws NoneType is not iterable error in discover method. When no Sonos is live on the network previously it returned None when calling discover.by_name api. But after upgrading to new release, instead returning None, its throws the error in soco library

Error log

```
user@mypc:~/Music$python play_local_files.py MySonos
Running as python 2
 Will use the following settings:
 Zone: MySonos
 IP of this machine: 172.21.2.156
 Use port: 8000
Traceback (most recent call last):
  File "play_local_files.py", line 159, in <module>
    main()
  File "play_local_files.py", line 126, in main
    zone = by_name(args.zone)
  File "/usr/local/lib/python2.7/dist-packages/soco/discovery.py", line 226, in by_name
    for device in discover():
TypeError: 'NoneType' object is not iterable
```

I am on python 2.7 and the soco installation information is below
user@mypc:~/Music$pip show soco
_Name: soco
Version: 0.17
Location: /usr/local/lib/python2.7/dist-packages
Requires: requests, xmltodict_

**Comment by KennethNielsen** [comment]
Hi @nmxprime 

Did you upgrade your speakers as well. ``discovery`` was changed for version 0.18, to accommodate a change in the Sonos API. But the change in SoCo is backwards incompatible, as outlined in the release notes: http://docs.python-soco.com/en/latest/releases/0.17.html)

Regards Kenneth

**Comment by jasontaylor190** [comment]
I have the same issue on Mac. Python 3.7.4, Soco 0.17, and Sonos Controller 10.3.

**Comment by KennethNielsen** [comment]
So it's been a while with no answer from @nmxprime, so I will close for now. Fell free to reopen if the issue persists.

@jasontaylor190 if you still have the problem please open a separate issue.
